bpf/kmesh/workload/include/tc.h

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Kmesh */
+
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+#include <linux/if_ether.h>
+#include <linux/in.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <linux/pkt_cls.h>
+#include <sys/socket.h>
+#include <stdbool.h>
+
+#include "bpf_log.h"
+#include "common.h"
+#include "ipsec_map.h"
+
+struct tc_info {
+    struct ethhdr *ethh;
+    union {
+        struct iphdr *iph;
+        struct ipv6hdr *ip6h;
+    };
+};
+
+#define PARSER_FAILED 1
+#define PARSER_SUCC   0
+
+static inline bool is_ipv4(struct tc_info *info)
+{
+    return info->ethh->h_proto == bpf_htons(ETH_P_IP);
+}
+
+static inline bool is_ipv6(struct tc_info *info)
+{
+    return info->ethh->h_proto == bpf_htons(ETH_P_IPV6);
+}
+
+
+static inline int parser_tc_info(struct __sk_buff *ctx, struct tc_info *info)
+{
+    void *begin = (void *)(long)(ctx->data);
+    void *end = (void *)(long)(ctx->data_end);
+
+    // eth header
```

Comment:
There is already a similar parse function (https://github.com/kmesh-net/kmesh/blob/main/bpf/kmesh/workload/xdp.c#L71), can it be reused or abstracted into a tool function.